stages/ui.cc

```diff
@@ -326,10 +340,17 @@ void Ui::UpdateLEDs() {
         }
         uint8_t type = configuration & 0x3;
         LedColor color = palette_[type];
+        uint16_t* seg_config = settings_->mutable_state()->segment_configuration;
         if (settings_->in_seg_gen_mode()) {
-          brightness = chain_state_->loop_status(i) == ChainState::LOOP_STATUS_SELF ?
-            lfo_patterns[configuration >> 8 & 0x3] : fade_patterns[chain_state_->loop_status(i)];
-
+          if (chain_state_->loop_status(i) == ChainState::LOOP_STATUS_SELF) {
+            brightness = lfo_patterns[configuration >> 8 & 0x3]; // triangular brightness pattern
+          } else if ((seg_config[i] & 0x3) == 0 && (seg_config[i] & 0x0300) == 0x0200) {
+            brightness = FadePattern(8, 0x08, 1); // slow ramp brightness pattern
```

Comment:
I think this one will be too slow if you add the solid delay beforehand. Probably just want to go down to 7, which will actually keep it at the same period if you add the delay.

Reply:
Also, you could add another array (like `lfo_patterns` and `fade_patterns`) so these are only calculated once, but I'm not sure it's worth it.